edx_sga/tests/test_sga.py

```diff
@@ -0,0 +1,617 @@
+        self.student = mock.Mock(username="fred6", is_staff=False, password="test")
+        self.state = '{"display_name": "Staff Graded Assignment"}'
+
+    def save(self):
+        """save method do nothing"""
+        pass
+
+
+@ddt
+class StaffGradedAssignmentMockedTests(unittest.TestCase):
+    """
+    Create a SGA block with mock data.
+    """
+    def setUp(self):
+        """
+        Creates a test course ID, mocks the runtime, and creates a fake storage
+        engine for use in all tests
+        """
+        super(StaffGradedAssignmentMockedTests, self).setUp()
+        self.course_id = CourseLocator(org='foo', course='baz', run='bar')
+        self.runtime = mock.Mock(anonymous_student_id='MOCK')
+        self.scope_ids = mock.Mock()
+        tmp = tempfile.mkdtemp()
+        patcher = mock.patch(
+            "edx_sga.sga.default_storage",
+            FileSystemStorage(tmp))
+        patcher.start()
+        self.addCleanup(patcher.stop)
+        self.staff = mock.Mock(return_value={
+            "password": "test",
+            "username": "tester",
+            "is_staff": True
+        })
+
+    def make_one(self, display_name=None, **kw):
+        """
+        Creates a XBlock SGA for testing purpose.
+        """
+        from edx_sga.sga import StaffGradedAssignmentXBlock as cls
+        field_data = DictFieldData(kw)
+        block = cls(self.runtime, field_data, self.scope_ids)
+        block.location = Location(
+            'foo', 'bar', 'baz', 'category', 'name', 'revision'
+        )
+
+        block.xmodule_runtime = self.runtime
+        block.course_id = self.course_id
+        block.scope_ids.usage_id = "i4x://foo/bar/category/name"
+        block.category = 'problem'
+
+        if display_name:
+            block.display_name = display_name
+
+        block.start = datetime.datetime(2010, 5, 12, 2, 42, tzinfo=pytz.utc)
+        return block
+
+    def test_ctor(self):
+        """
+        Test points are set correctly.
+        """
+        block = self.make_one(points=10)
+        self.assertEqual(block.display_name, "Staff Graded Assignment")
+        self.assertEqual(block.points, 10)
+
+    def test_max_score(self):
+        """
+        Text max score is set correctly.
+        """
+        block = self.make_one(points=20)
+        self.assertEqual(block.max_score(), 20)
+
+    def test_max_score_integer(self):
+        """
+        Test assigning a float max score is rounded to nearest integer.
+        """
+        block = self.make_one(points=20.4)
+        self.assertEqual(block.max_score(), 20)
+
+    def personalize_upload(self, block, upload):
+        # pylint: disable=unused-argument
+        """
+        Set values on block from file upload.
+        """
+        now = datetime.datetime.utcnow().replace(tzinfo=pytz.timezone(getattr(settings, "TIME_ZONE", pytz.utc.zone)))
+        setattr(block, "annotated_mimetype", mimetypes.guess_type(upload.file.name.encode('utf-8'))[0])
```

Comment:
Is it possible to do `block.annotated_mimetype = mimetypes.guess_type(upload.file.name.encode('utf-8'))[0]` instead? Can we do this in the next few lines too?